### PR TITLE
Added DomainTag changes to ReadHistory branch for readHistoryRequest, CreateWorkflowRequest + added DomainCacheNoOp file

### PR DESCRIPTION
**Pull request by abhishekj720**
**What changed?**
Added DomainTag changes to ReadHistory branch for readHistoryRequest, CreateWorkflowRequest methods in persistenceMetricsClients + added domainCacheNoOp

**Why?**
To create metrics based on Domain Tag

**How did you test it?**
Tested Locally, added mock tests

**Release notes**
N/A

**Documentation Changes**
N/A

### CODE REVIEW

File: common/reconciliation/invariant/concreteExecutionExists.go [hunk 1]
```diff
@@ -34,15 +35,17 @@ import (
...
 )
 
 // NewConcreteExecutionExists returns a new invariant for checking concrete execution
 func NewConcreteExecutionExists(
-	pr persistence.Retryer,
+	pr persistence.Retryer, dc cache.DomainCache,
 ) Invariant {
 	return &concreteExecutionExists{
 		pr: pr,
+		dc: dc,
```
Comment (Shaddoll) [inline common/reconciliation/invariant/concreteExecutionExists.go:48]: domainCache is added but not used.
The same thing applies to other files in this package.

File: common/reconciliation/invariant/timerInvalid.go [hunk 2]
```diff
@@ -25,13 +25,15 @@ package invariant
...
 
+	// "github.com/uber/cadence/common/cache"
 	"github.com/uber/cadence/common/persistence"
 	"github.com/uber/cadence/common/reconciliation/entity"
 	"github.com/uber/cadence/common/types"
 )
 
 type TimerInvalid struct {
 	pr persistence.Retryer
+	// ch cache.DomainCache
```
Comment (Shaddoll) [inline common/reconciliation/invariant/timerInvalid.go:36]: This is not implemented

File: service/worker/scanner/timers/timers.go [hunk 3]
```diff
@@ -147,6 +149,7 @@ func FixerManager(
 	_ context.Context,
 	pr persistence.Retryer,
 	_ shardscanner.FixShardActivityParams,
+	cache cache.DomainCache,
```
Comment (Shaddoll) [inline service/worker/scanner/timers/timers.go:152]: cache is not used in this function, is it required?

File: common/cache/domainCacheNoOp.go [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+func (c *noOpDomainCache) GetDomainID(
+	name string,
+) (string, error) {
+	return "", errors.New("Not supported for noOpDomainCache")
+}
+
+func (c *noOpDomainCache) GetDomainName(
+	id string,
+) (string, error) {
+	return "", errors.New("Not supported for noOpDomainCache")
```
Comment (abhishekj720) [inline common/cache/domainCacheNoOp.go:105]: return "",nil

File: common/persistence/dataManagerInterfaces.go [hunk 5]
```diff
@@ -956,6 +960,8 @@ type (
 		NewWorkflowSnapshot *WorkflowSnapshot
 
 		Encoding common.EncodingType // optional binary encoding type
+
+		DomainName string
```
Comment (Shaddoll) [inline common/persistence/dataManagerInterfaces.go:964]: The metrics of UpdateWorkflowExecution is not updated

File: common/persistence/persistence-tests/persistenceTestBase.go [hunk 6]
```diff
@@ -364,7 +365,8 @@ func (s *TestBase) CreateWorkflowExecutionWithBranchToken(
 			Checksum:         testWorkflowChecksum,
 			VersionHistories: versionHistories,
 		},
-		RangeID: s.ShardInfo.RangeID,
+		RangeID:    s.ShardInfo.RangeID,
+		DomainName: s.DomainManager.GetName(),
```
Comment (Shaddoll) [inline common/persistence/persistence-tests/persistenceTestBase.go:369]: s.DomainManager.GetName() is not the domain name here.
BTW, since this is only in test, you can leave this empty.

File: common/reconciliation/invariant/openCurrentExecution.go [hunk 7]
```diff
@@ -34,15 +35,17 @@ import (
...
 	openCurrentExecution struct {
 		pr persistence.Retryer
+		dc cache.DomainCache
 	}
 )
 
 // NewOpenCurrentExecution returns a new invariant for checking open current execution
 func NewOpenCurrentExecution(
-	pr persistence.Retryer,
+	pr persistence.Retryer, dc cache.DomainCache,
```
Comment (Shaddoll) [inline common/reconciliation/invariant/openCurrentExecution.go:44]: I think you'll make the corresponding change in another PR? Do you mind adding a TODO here?

File: common/reconciliation/invariant/openCurrentExecution_test.go [hunk 8]
```diff
@@ -174,7 +176,10 @@ func (s *OpenCurrentExecutionSuite) TestCheck() {
 		execManager := &mocks.ExecutionManager{}
 		execManager.On("GetWorkflowExecution", mock.Anything, mock.Anything).Return(tc.getConcreteResp, tc.getConcreteErr)
 		execManager.On("GetCurrentExecution", mock.Anything, mock.Anything).Return(tc.getCurrentResp, tc.getCurrentErr)
-		o := NewOpenCurrentExecution(persistence.NewPersistenceRetryer(execManager, nil, c2.CreatePersistenceRetryPolicy()))
+
+		ctrl := gomock.NewController(s.T())
+		domainCache := cache.NewMockDomainCache(ctrl)
```
Comment (Shaddoll) [inline common/reconciliation/invariant/openCurrentExecution_test.go:181]: I think this is not needed, since you didn't invoke domain cache method

File: service/worker/scanner/executions/types.go [hunk 9]
```diff
@@ -109,7 +110,7 @@ func (st ScanType) ToInvariants(collections []invariant.Collection) []InvariantF
 		for _, collection := range collections {
 			switch collection {
 			case invariant.CollectionMutableState:
-				fns = append(fns, invariant.NewConcreteExecutionExists)
```
Comment (Shaddoll) [inline service/worker/scanner/executions/types.go:112]: this change doesn't seem to be expected

File: common/reconciliation/invariant/openCurrentExecution.go (hunk 7 above)
Comment (abhishekj720) [reply common/reconciliation/invariant/openCurrentExecution.go:44]: but it has been used for fns []InvariantFactory

fns = append(fns, invariant.NewOpenCurrentExecution)

File: common/reconciliation/invariant/timerInvalid.go [hunk 10]
```diff
@@ -61,7 +61,6 @@ func (h *TimerInvalid) Check(
 			Info:            "failed to check: expected timer entity",
 		}
 	}
-
```
Comment (Shaddoll) [inline common/reconciliation/invariant/timerInvalid.go:64]: I think you'll need to do the same domainCache change here as in historyExists.go and openCurrenntExecution.go.
Are you planning to do in another PR?

File: common/persistence/persistence-tests/persistenceTestBase.go [hunk 11]
```diff
@@ -439,7 +440,8 @@ func (s *TestBase) CreateChildWorkflowExecution(ctx context.Context, domainID st
 			TimerTasks:       timerTasks,
 			VersionHistories: versionHistories,
 		},
-		RangeID: s.ShardInfo.RangeID,
+		RangeID:    s.ShardInfo.RangeID,
+		DomainName: s.DomainManager.GetName(),
```
Comment (Shaddoll) [inline common/persistence/persistence-tests/persistenceTestBase.go:444]: nit: The DomainName set here is not the correct value. But since this is the test code, and we don't care about the metrics in test, it's ok to not add domainName in test code.

**Review by Shaddoll — APPROVED** [review]
